File: Clients/BlogBlazor/Shared/Blazor.Contracts/V1/Validators/UsersValidators/LoginRequestValidator.cs
using Blazor.Contracts.V1.Requests.UsersRequests;
using FluentValidation;

namespace Blazor.Contracts.V1.Validators.UsersValidators
{
    /// <summary>
    /// Login request validator.
    /// </summary>
    /// <seealso cref="AbstractValidator{LoginRequest}" />
    public class LoginRequestValidator : AbstractValidator<LoginRequest>
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="CreateTagRequestValidator"/> class.
        /// </summary>
        public LoginRequestValidator()
        {
            RuleFor(x => x.Email)
                .NotEmpty()
                .EmailAddress();

            RuleFor(x => x.Password)
                .NotEmpty()
                .MinimumLength(6)
                .Matches("^[a-zA-Z0-9-_]*S");
        }
    }
}

File: Shared/Blog.FSharp.Contracts/V1/Validators/UsersValidators/LoginRequestValidator.fs
namespace Blog.FSharp.Contracts.V1.Validators.UsersValidators

open FluentValidation
open Blog.FSharp.Contracts.V1.Requests.UsersRequests

/// <summary>
/// Validator for login request.
/// </summary>
type LoginRequestValidator() =
    inherit AbstractValidator<LoginRequest>()

    do
        // Validate Email: not empty, valid email format
        base.RuleFor(fun x -> x.Email)
            .NotEmpty()
            .EmailAddress() |> ignore

        // Validate Password: not empty, at least 6 chars, matches regex
        base.RuleFor(fun x -> x.Password)
            .NotEmpty()
            .MinimumLength(6)
            .Matches("^[a-zA-Z0-9-_]*S") |> ignore